coalib/results/Diff.py

```diff
@@ -278,18 +278,37 @@ def modified(self):
         return result
 
     @property
+    def modified(self):
+        """
+        Calculates the modified file, after applying the Diff to the original.
+
+        This property also adds linebreaks at the end of each line.
+        If no newline was present at the end of file before, this state will
+        be preserved, except if the last line is deleted.
+        """
+        return self._generate_linebreaks(self._raw_modified())
+
+    @property
     def unified_diff(self):
         """
         Generates a unified diff corresponding to this patch.
 
+        Each change will be displayed on its own line. Additionally, the
+        unified diff preserves the EOF-state of the original file. This
+        means that the ``Diff`` will only have a linebreak on the last line,
+        if that was also present in the original file.
+
         Note that the unified diff is not deterministic and thus not suitable
         for equality comparison.
         """
-        return ''.join(difflib.unified_diff(
-            self.original,
-            self.modified,
+
+        list_unified_diff = list(difflib.unified_diff(
+            self._file,
+            self._raw_modified(),
             tofile=self.rename if isinstance(self.rename, str) else ''))
 
+        return ''.join(self._generate_linebreaks(list_unified_diff))
```

Comment:
Yes, but in case of a file without linebreaks , this would print all changes in one line.
`diff = Diff(['1','2'])`
`diff.modify_line(1,'2')`
`diff.modify_line(2,'1')`
`diff.unified_diff => '--- \n+++ \n@@ -1,2 +1,2 @@\n+2 1-2'`

In a case where only the last line does not have a linebreak it would look even weirder:

`diff = Diff(['1\n','2'])
diff.modify_line(1,'2')
diff.modify_line(2,'1')
diff.unified_diff => '--- \n+++ \n@@ -1,2 +1,2 @@\n-1\n 2+1'`

(I guess the ordering change is due to the unified diff being non-deterministic?)